src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,1077 @@
+        self.assertEqual('Installed', str(message["patches"][1]["patchInstallationState"]))
+        self.assertEqual(message["patches"][2]["name"], "python-samba2")
+        self.assertEqual('python-samba0_2:4.4.5+dfsg-2ubuntu5.4_Ubuntu_16.04', str(message["patches"][0]["patchId"]))
+        self.assertTrue('Critical' in str(message["patches"][2]["classifications"]))
+
+        # Clean up complete.status files
+        self.runtime.env_layer.file_system.delete_files_from_dir(self.runtime.status_handler.status_file_path, '*.complete.status')
+
+    def __assert_no_truncation_status_file(self, patch_summary, status, patch_count):
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        message = json.loads(substatus_file_data[0]["status"]["substatus"][0]["formattedMessage"]["message"])
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.assertEqual(substatus_file_data[0]["status"]["substatus"][0]["name"], patch_summary)
+        self.assertEqual(substatus_file_data[0]["status"]["substatus"][0]["status"], status.lower())
+        self.assertEqual(len(message["patches"]), patch_count)
+        self.assertNotEqual(message["patches"][-1]['patchId'], "Truncated_patch_list_id")
+        self.assertTrue('Truncated_patch_list_id' not in message["patches"][-1]['name'])
+        self.assertEqual(message["errors"]["code"], 0)
+        self.assertEqual(len(message["errors"]["details"]), 0)
+        self.assertFalse("review this log file on the machine" in message)
+
+    def __assert_patch_summary_from_status(self, substatus_file_data, operation, patch_summary, status, substatus_index=0, is_under_agent_size_limit=True):
+        self.assertEqual(substatus_file_data[0]["status"]["operation"], operation)
+
+        if is_under_agent_size_limit:
+            # Assert status file size < 128kb n 126kb
+            truncated_substatus_file_byte_size = len(json.dumps(substatus_file_data))
+            self.assertTrue(truncated_substatus_file_byte_size < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+            self.assertTrue(truncated_substatus_file_byte_size < Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)
+        else:
+            # Assert complete status file size > 128kb
+            self.assertTrue(len(json.dumps(substatus_file_data)) > Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+
+        if patch_summary == Constants.PATCH_ASSESSMENT_SUMMARY:
+            substatus_file_data = substatus_file_data[0]["status"]["substatus"][substatus_index]
+        else:
+            substatus_file_data = substatus_file_data[0]["status"]["substatus"][substatus_index]
+
```

Comment:
There's no need for this if else block, since both do the same thing

Reply:
done